### ISSUE TITLE
Chapter 02 Linear regression from scratch - mistake in the code

### ISSUE THREAD

**Issue by AndreasHoermandinger** [zackchase/mxnet-the-straight-dope#397]
Hi, I found a possible error in the code, at least it didn't run for me  - mxnet 0.12, python 3.6.3
In code snippet 41 the model paramenters `w` and `b` are reinitialised, but gradients are not attached which would make the script crash at `loss.backward()`, as `autograd` is not building any computational graph. To fix you would need to add calls `attach_grad()` for all parameters, like you did in snippets 35, 36:
```
w = nd.random_normal(shape=(num_inputs, num_outputs), ctx=model_ctx)
b = nd.random_normal(shape=num_outputs, ctx=model_ctx)
params = [w,b]

for param in params:
    param.attach_grad()
```
